CALLS.md

```diff
@@ -0,0 +1,62 @@
+Calls
+=====
+
+The HTTP client’s job is to accept your request and produce its response. This is simple in theory but it gets tricky in practice.
+
+#### [Requests](http://square.github.io/okhttp/3.x/okhttp/okhttp3/Request.html)
+
+Each HTTP request contains a URL, a method (like `GET` or `POST`), and a list of headers. Requests may also contain a body: a data stream of a specific content type.
+
+#### [Responses](http://square.github.io/okhttp/3.x/okhttp/okhttp3/Response.html)
+
+The response answers the request with a code (like 200 for success or 404 for not found), headers, and its own optional body.
+
+#### Rewriting Requests
+
+When you provide OkHttp with an HTTP request, you’re describing the request at a high-level: _“fetch me this URL with these headers.”_ For correctness and efficiency, OkHttp rewrites your request before transmitting it.
+
+OkHttp may add headers that are absent from the original request, including `Content-Length`, `Transfer-Encoding`, `User-Agent`, `Host`, `Connection`, and `Content-Type`. It will add an `Accept-Encoding` header for transparent response compression unless the header is already present. If you’ve got cookies, OkHttp will add a `Cookie` header with them.
+
+Some requests will have a cached response. When this cached response isn’t fresh, OkHttp can do a _conditional GET_ to download an updated response if it’s newer than what’s cached. This requires headers like `If-Modified-Since` and `If-None-Match` to be added.
+
+#### Rewriting Responses
+
+If transparent compression was used, OkHttp will drop the corresponding response headers `Content-Encoding` and `Content-Length` because they don’t apply to the decompressed response body.
+
+If a conditional GET was successful, responses from the network and cache are merged as directed by the spec.
```

Comment:
@swankjesse Should we specify or add link for the said spec in case anyone is interested? 

Was trying to dig into the code to locate it with no luck 😅. Can you or someone point me to where can I find it in the code if available?

Reply:
https://tools.ietf.org/html/rfc7234